File: javascript/hamming/hamming.js
function range(n) {
  return Array.apply(null, Array(n)).map(function(_, i) { return i; });
}

function zip(a, b) {
  return range(a.length).map(function(i) { return [a[i], b[i]] });
}

var Hamming = function() {};

Hamming.prototype.compute = function(a, b) {
    if (a.length != b.length) {
      throw new Error('DNA strands must be of equal length.');
    }

    return zip(a, b).reduce(function(acc, pair) {
      var x = pair[0];
      var y = pair[1];
      return x == y ? acc : acc + 1;
    }, 0);
};

module.exports = Hamming;

File: kotlin/hamming/src/main/kotlin/Hamming.kt
const val LENGTH_INEQUALITY = "left and right strands must be of equal length."

object Hamming {
  fun compute(a: String, b: String): Int {
    if (a.length == b.length)
      return a.zip(b).count { (x, y) -> x != y }
    else
      throw IllegalArgumentException(LENGTH_INEQUALITY)
  }
}
